plugins/scripting/src/scriptingtool.cpp

```diff
@@ -0,0 +1,193 @@
+
+	m_tool->addWidgetToTopContainerHelper(loadBtn, TTA_LEFT);
+	m_tool->addWidgetToTopContainerHelper(saveBtn, TTA_LEFT);
+	m_tool->addWidgetToTopContainerHelper(toggleTerminalMode, TTA_RIGHT);
+	m_tool->addWidgetToTopContainerHelper(m_runBtn, TTA_RIGHT);
+
+	m_codeEditor = new ScopyCodeEditor(m_tool);
+	Style::setStyle(m_codeEditor, style::properties::widget::basicComponent);
+	Style::setStyle(m_codeEditor, style::properties::widget::border_interactive);
+
+	m_codeOutput = new QPlainTextEdit(m_tool);
+	m_codeOutput->setReadOnly(true);
+	Style::setStyle(m_codeOutput, style::properties::widget::basicComponent);
+	Style::setStyle(m_codeOutput, style::properties::widget::border_interactive);
+
+	connect(loadBtn, &QPushButton::clicked, this, [=]() { loadFile(); });
+	connect(saveBtn, &QPushButton::clicked, this, [=]() { saveToFile(); });
+	connect(m_runBtn, &QPushButton::clicked, this, [=]() { compileCode(m_codeEditor->toPlainText()); });
+
+	QLineEdit *lineEdit = new QLineEdit(m_tool);
+	lineEdit->setVisible(false);
+	Style::setStyle(lineEdit, style::properties::widget::border_interactive);
+	Style::setStyle(lineEdit, style::properties::widget::basicComponent);
+
+	connect(lineEdit, &QLineEdit::returnPressed, this, [=]() {
+		QString input = lineEdit->text();
+		compileCode(input);
+		m_codeEditor->appendPlainText(input);
+		lineEdit->clear();
+	});
+
+	connect(toggleTerminalMode->onOffswitch(), &QPushButton::toggled, this, [=](bool toggled) {
+		lineEdit->setVisible(toggled);
+		// when enabling console mode clear code
+		if(toggled)
+			m_codeEditor->clear();
+	});
+
+	m_tool->addWidgetToCentralContainerHelper(m_codeEditor);
+	m_tool->addWidgetToCentralContainerHelper(m_codeOutput);
+	m_tool->addWidgetToCentralContainerHelper(lineEdit);
+}
+
+void ScriptingTool::loadFile()
+{
+	// chose file
+	QString selectedFilter;
+	QString fileName = QFileDialog::getSaveFileName(this, tr("Export"), "", tr("All Files(*)"), &selectedFilter,
+							QFileDialog::Options(QFileDialog::DontUseNativeDialog));
+
+	// display file content
+	QFile file(fileName);
+
+	if(!file.isOpen()) {
+		file.open(QIODevice::ReadOnly);
+
+		QTextStream in(&file);
+		QString fileContent = "";
+		while(!in.atEnd()) {
+			QString line = in.readLine();
+			fileContent += line + "\n";
+		}
+		m_codeEditor->setPlainText(fileContent);
+	} else {
+		qDebug() << "File already opened! ";
+		m_codeOutput->setPlainText("File already opened! ");
+	}
+
+	if(file.isOpen())
+		file.close();
+}
+
+void ScriptingTool::saveToFile()
+{
+	QString selectedFilter;
+	QString fileName = QFileDialog::getSaveFileName(this, tr("Export"), "", tr("All Files(*)"), &selectedFilter,
+							QFileDialog::Options(QFileDialog::DontUseNativeDialog));
+
+	// display file content
+	QFile file(fileName);
+
+	if(!file.isOpen()) {
+		file.open(QIODevice::WriteOnly);
+		QTextStream out(&file);
+		out << m_codeEditor->toPlainText();
+
+		m_codeOutput->setPlainText("Save completed");
```

Comment:
We could print a timestamp for each line in the code output, this way it's easier to follow the log.